packages/cli/src/commands/versions/bump.ts

```diff
@@ -284,27 +311,89 @@ export default async (cmd: Command) => {
   }
 };
 
-function createVersionFinder() {
-  const found = new Map<string, string>();
+export function createStrictVersionFinder(options: {
+  releaseManifest: ReleaseManifest;
+}) {
+  const releasePackages = new Map(
+    options.releaseManifest.packages.map(p => [p.name, p.version]),
+  );
+  return async function findTargetVersion(name: string) {
+    console.log(`Checking for updates of ${name}`);
+    const manifestVersion = releasePackages.get(name);
+    if (manifestVersion) {
+      return manifestVersion;
+    }
+    throw new NotFoundError(`Package ${name} not found in release manifest`);
```

Comment:
What happens when you have packages that were added/removed across release lines? Like, if people depend on `core-plugin-api` and we refactor to instead call it `core-frontend-plugin-api` or whatever. Will all line based bumps just fail then? I mean, in a sense it might be the right thing to do since there's no sensible course of action that doesn't require user intervention, but just checking

Reply:
The current behaviour is to warn and skip if the package name is not found which I think is what we want for now.

https://github.com/backstage/backstage/pull/9092/files#diff-8fd03dee306a059c2d5dfd107f72975a7f9800edb53fee3df7827f9a2d9b28ccR102